File: backend/domain/dish.go
package domain

type Dish struct {
	Id          int32
	Name        string
	Description string
	Price       int32
	Url         string   `json:",omitempty"`
	Categories  []string `json:",omitempty"`
}

type GetDishesRequest struct {
	Ids           string `query:"ids"`
	CategoriesIds string `query:"categoriesIds"`
	Limit         int32  `query:"limit" validate:"max=30"`
	Offset        int32  `query:"offset"`
}

type AddDishRequest struct {
	Name        string `validate:"required,min=1"`
	Description string
	Price       int32   `validate:"gt=0"`
	Categories  []int32 `json:",omitempty"`
	Image       []byte  `json:",omitempty"`
}

File: frontend/src/client/dish.ts
import { DefaultClient } from '../utils/client'
import { GetAccessToken } from './user'

export class Dish {
    id: number
    name: string
    description: string
    categories: string
    url: string
    price: number
}

const dishesEndpoint = '/dishes'
export async function GetDishes(dishIds?: any[], limit?, offset?, categoriesIds?) {
    let queryParams = {}
    if (dishIds && dishIds.length > 0) {
        queryParams = { "ids": dishIds.join(',') }
    } else {
        queryParams = { "limit": limit, offset: offset }
        if (categoriesIds) queryParams["categoriesIds"] = categoriesIds
    }
    return await DefaultClient.Get(dishesEndpoint, queryParams).
        then(response => response.json()).
        catch(reason => console.log(reason))
}

export class AddDishObj {
    name: string
    description: string
    price: number
    categories: number[]
    image: any
}

export async function AddDish(dish: AddDishObj): Promise<boolean | void> {
    const accessToken = await GetAccessToken();
    return await DefaultClient.PostJSON(dishesEndpoint, dish, DefaultClient.UserBearerAuthHeader(accessToken)).
        then(resp => resp.ok).
        catch(reason => alert(reason))
}

export async function DeleteDish(id: any) {
    const accessToken = await GetAccessToken();
    return await DefaultClient.Delete(dishesEndpoint + "/delete/" + id, null, DefaultClient.UserBearerAuthHeader(accessToken)).
        then(resp => resp.ok).
        catch(reason => alert(reason))
}

export async function EditDish(dish: Dish) {
    const accessToken = await GetAccessToken();
    return await DefaultClient.PostJSON(dishesEndpoint + "/edit/" + dish.id, dish, DefaultClient.UserBearerAuthHeader(accessToken)).
        then(resp => resp.ok).
        catch(reason => alert(reason))
}